src/util/mpir_nodemap.h

```diff
@@ -322,13 +322,83 @@ static inline int MPIR_NODEMAP_populate_ids_from_mapping(char *mapping,
 
    Each process kvs_puts its hostname and stores the total number of
    processes (g_num_global).  Each process determines maximum node id
-   (g_max_node_id) and assigns a node id to each process (g_node_ids[]):
+   and assigns a node id to each process (g_node_ids[]):
 
      For each hostname the process seaches the list of unique nodes
      names (node_names[]) for a match.  If a match is found, the node id
      is recorded for that matching process.  Otherwise, the hostname is
      added to the list of node names.
 */
+
+/* TODO: make the fallback routine general, ie works across all PMI versions */
```

Comment:
Does this mean the PR is not feature complete yet?

Reply:
"TODO" for another PR :)  Currently the fallback in needed for `gforker` and `remshell`, which only supports PMIv1. However, the fallback mechanism is general enough that it can be implemented for all PMI versions in the same way. By making it general, the code can be cleaner and handy when we adding new PMI implementations (in the future).

Reply:
Exactly what I wanted to know :)